src/asm2wasm.h

```diff
@@ -1550,6 +1272,7 @@ void Asm2WasmBuilder::processAsm(Ref ast) {
   // finalizeCalls also does autoDrop, which is crucial for the non-optimizing case,
   // so that the output of the first pass is valid
   passRunner.add<FinalizeCalls>(this);
+  addTrapModePass(passRunner, trapMode);
```

Comment:
this will run that pass after the functions have already been (mostly) optimized.